src/qililab/instruments/qblox/qblox_qrm.py

```diff
@@ -139,12 +139,14 @@ def _get_qprogram_acquisitions(self, acquisitions: list[str]) -> list[QbloxQProg
                         sequencer=sequencer.identifier,
                         timeout=cast(AWGQbloxADCSequencer, sequencer).acquisition_timeout,
                     )
+                    self.device.store_scope_acquisition(sequencer=sequencer.identifier, name=acquisition)
                     raw_measurement_data = self.device.get_acquisitions(sequencer=sequencer.identifier)[acquisition][
                         "acquisition"
                     ]
                     measurement_result = QbloxQProgramMeasurementResult(raw_measurement_data=raw_measurement_data)
                     results.append(measurement_result)
-
+        for sequencer in self.awg_sequencers:
+            self.device.delete_acquisition_data(sequencer=sequencer.identifier, all=True)
```

Comment:
```suggestion
        
                self.device.delete_acquisition_data(sequencer=sequencer.identifier, all=True)
```
can we just do this inside the `sequencer in self.awg_sequencers` loop above?

Reply:
No, because the sequencer loop above is inside the acquisitions loop, so the command will be executed multiple times for each sequencer. Plus, if you do `delete_acquisition_data(sequencer=sequencer.identifier, all=True)` inside that loop you will delete all data before retrieving it with `store_scope_acquisition(sequencer=sequencer.identifier, name=acquisition)`. So you only want to delete the data after retrieving everything.

Reply:
No, because the sequencer loop above is inside the acquisitions loop, so the command will be executed multiple times for each sequencer. Plus, if you do `delete_acquisition_data(sequencer=sequencer.identifier, all=True)` inside that loop you will delete all data before retrieving it with `store_scope_acquisition(sequencer=sequencer.identifier, name=acquisition)`. So you only want to delete the data after retrieving everything.

Reply:
fair enough, why not switch the order of loops then?

```python
        for sequencer in self.awg_sequencers:
            for acquisition in acquisitions:
                if sequencer.identifier in self.sequences:
                    self.device.get_acquisition_state(
                        sequencer=sequencer.identifier,
                        timeout=cast(AWGQbloxADCSequencer, sequencer).acquisition_timeout,
                    )
                    self.device.store_scope_acquisition(sequencer=sequencer.identifier, name=acquisition)
                    raw_measurement_data = self.device.get_acquisitions(sequencer=sequencer.identifier)[acquisition][
                        "acquisition"
                    ]
                    measurement_result = QbloxQProgramMeasurementResult(raw_measurement_data=raw_measurement_data)
                    results.append(measurement_result)

            self.device.delete_acquisition_data(sequencer=sequencer.identifier, all=True)
```
since `self.awg_sequencers` is always the same (it's a class attribute) switching the order of the loops should be the same

Reply:
this is quite trivial though, if you prefer having the delete data later in a separate loop I don't think it's that different performance wise

Reply:
There is no performance drawback of having a linear loop, especially when it has maximum 6 elements 😅